packages/features/bookings/Booker/components/AvailableTimeSlots.tsx

```diff
@@ -34,30 +44,31 @@ export const AvailableTimeSlots = ({ extraDays, limitHeight, seatsPerTimeslot }:
   };
 
   const schedule = useScheduleForEvent({
-    prefetchNextMonth: !!extraDays && dayjs(date).month() !== dayjs(date).add(extraDays, "day").month(),
+    prefetchNextMonth,
+    monthCount,
   });
+  const nonEmptyScheduleDays = useNonEmptyScheduleDays(schedule?.data?.slots);
 
   // Creates an array of dates to fetch slots for.
   // If `extraDays` is passed in, we will extend the array with the next `extraDays` days.
   const dates = useMemo(
     () =>
       !extraDays
         ? [date]
-        : [
-            // If NO date is selected yet, we show by default the upcomming `nextDays` days.
-            date,
-            ...Array.from({ length: extraDays }).map((_, index) =>
-              dayjs(date)
-                .add(index + 1, "day")
-                .format("YYYY-MM-DD")
-            ),
-          ],
+        : nonEmptyScheduleDays.length > 0 
+          ? nonEmptyScheduleDays.slice(sliceFrom, sliceTo)
+          :[],
     [date, extraDays]
   );
-
-  const isMultipleDates = dates.length > 1;
-  const slotsPerDay = useSlotsForMultipleDates(dates, schedule?.data?.slots);
-
+  
+  const slotsPerDay = schedule?.data?.slots 
+    ? useSlotsForAvailableDates(dates, isColumnView 
+      ? Object.values(schedule.data.slots).slice(sliceFrom, sliceTo) 
+      : nonEmptyScheduleDays.includes(date) 
+        ? Object.values(schedule.data.slots).slice(nonEmptyScheduleDays.indexOf(date)) 
+        : [] ) 
+    : useSlotsForAvailableDates(dates, []);
+  
```

Comment:
can you simplify this code? This uses a lot of ternary operators

Reply:
since month view and column view uses same function. And we need to differentiate those. so these much is required..

Reply:
Hope the update make sense.